pytermgui/ansi_interface.py

```diff
@@ -107,243 +142,263 @@ class Color:
+            self._OldStdoutMode = ctypes.wintypes.DWORD()
+            kernel32.GetConsoleMode(
+                kernel32.GetStdHandle(-10), ctypes.byref(self._OldStdinMode)
+            )
+            kernel32.GetConsoleMode(
+                kernel32.GetStdHandle(-11), ctypes.byref(self._OldStdoutMode)
+            )
+        else:
+            self._OldStdinMode = termios.tcgetattr(sys.stdin)
 
-        raise
+    @contextmanager
+    def NoEcho(self) -> None:
+        if _IS_NT:
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), 0)
+            kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), 7)
+        else:
+            _ = self._OldStdinMode
+            _[3] = _[3] & ~(termios.ECHO | termios.ICANON)
+            termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, _)
+        try:
+            yield
+        finally:
+            if _IS_NT:
+                kernel32.SetConsoleMode(kernel32.GetStdHandle(-10), self._OldStdinMode)
+                kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), self._OldStdoutMode)
+            else:
+                termios.tcsetattr(sys.stdin, termios.TCSAFLUSH, self._OldStdinMode)
+
+
+terminal_modes = TerminalModes()
+"""
+TerminalModes instance
+"""
 
 
-class _Terminal:
-    """A class to store & access data about a terminal"""
+class Terminal:
+    """
+    A class to store & access data about a terminal
+    """
 
     RESIZE = 0
     margins = [0, 0, 0, 0]
 
     def __init__(self) -> None:
-        """Initialize object"""
-
+        """
+        Initialize 'Terminal' class
+        """
         self.origin: tuple[int, int] = (1, 1)
-        self.size: tuple[int, int] = self._get_size()
+        self.size: tuple[int, int] = get_terminal_size()
         self._listeners: dict[int, list[Callable[..., Any]]] = {}
-
         # TODO: Windows doesn't support SIGWINCH, is there another alternative?
-        if hasattr(signal, "SIGWINCH"):
+        if _IS_NT:
+            asyncio.run(self._WinSIGWINCH())
+        else:
             signal.signal(signal.SIGWINCH, self._update_size)
 
-    def _call_listener(self, event: int, data: Any) -> None:
-        """Call event callback is one is found."""
+    async def _WinSIGWINCH(self) -> None:
+        while True:
+            n_width = get_terminal_size()
+            if n_width != self.size:
+                self._update_size(28, (sys._getframe(1) if _SYS_HAS_FRAME else None))
+            await asyncio.sleep(0.5)
 
+    def _call_listener(self, event: int, data: Any) -> None:
+        """
+        Call event callback is one is found.
+        """
         if event in self._listeners:
             for callback in self._listeners[event]:
                 callback(data)
 
-    def _get_size(self) -> tuple[int, int]:
-        """Get screen size while substracting the origin position"""
-
-        # This always has len() == 2, but mypy can't see that.
-        return tuple(val - org for val, org in zip(screen_size(), self.origin))  # type: ignore
```

Comment:
The `origin` calculations here are not done if `get_terminal_size` is used. You should **not** remove this.